pkg/controller/podautoscaler/metrics/metrics_client.go

```diff
@@ -44,11 +44,11 @@ type MetricsClient interface {
 	// GetCPUUtilization returns the average utilization over all pods represented as a percent of requested CPU
 	// (e.g. 70 means that an average pod uses 70% of the requested CPU)
 	// and the time of generation of the oldest of utilization reports for pods.
-	GetCPUUtilization(namespace string, selector map[string]string) (*int, time.Time, error)
+	GetCPUUtilization(namespace string, selector string) (*int, time.Time, error)
```

Comment:
Daniel is correct. I'd like to add string selectors to the status of services and all controllers so that clients don't need to understand the selector structure in order to construct a list query. 

Our clients do too much decoding and encoding, which obstructs evolution and extensibility, and makes it harder to write clients in other languages.